File: src/main/java/org/codinjutsu/tools/nosql/commons/view/add/NumberFieldWrapper.java
package org.codinjutsu.tools.nosql.commons.view.add;

import org.codinjutsu.tools.nosql.commons.model.internal.layer.DatabaseElement;
import org.codinjutsu.tools.nosql.commons.view.nodedescriptor.internal.InternalDatabasePrimitive;
import org.jetbrains.annotations.NotNull;

import static org.codinjutsu.tools.nosql.commons.utils.StringUtilsKt.parseNumber;

public class NumberFieldWrapper extends JTextFieldWrapper {

    @Override
    public DatabaseElement getValue() {
        return new InternalDatabasePrimitive(parseValue());
    }

    @NotNull
    private Number parseValue() {
        return parseNumber(component.getText());
    }

    @Override
    public void validate() {
        super.validate();
        parseValue();
    }
}

File: src/main/java/org/codinjutsu/tools/nosql/commons/view/add/NumberFieldWrapper.kt
package org.codinjutsu.tools.nosql.commons.view.add

import org.codinjutsu.tools.nosql.commons.model.internal.layer.impl.DatabasePrimitiveImpl
import org.codinjutsu.tools.nosql.commons.utils.parseNumber

internal class NumberFieldWrapper : JTextFieldWrapper() {

    override fun getValue() = DatabasePrimitiveImpl(parseValue())

    private fun parseValue() = component.text.parseNumber()

    override fun validate() {
        super.validate()
        parseValue()
    }
}
